schema/sql/schema.sql

```diff
@@ -97,6 +97,14 @@ create table "ssh_public_key" (
   "uid" integer references passwd (uid) on delete cascade
 );
 
+create domain pgp_sha256_fingerprint bytea check (length(value) = 64);
+
+create table "gpg_public_key" (
+  "uid" integer references passwd (uid) on delete cascade,
+  "fingerprint" pgp_sha256_fingerprint not null,
+  "key" bytea not null check (length(key) < 2097152) -- max 2M
```

Comment:
Is this the raw public key or is this the key along with identity information attached? Ideally, I'd prefer the second. I think a good alternative as well is to just have something that is as close to an "ASCII armored" key using the GPG `--export-options=export-minimal` (I think) flag.

Reply:
So it really depends on what we want to do here. We can either hold the entire public key. Or we can do our own trimming to only include the UIDs and necessary signature packets. Similar to what hagrid does.

I can see a couple of things around this
1) Having the entire key is just simple
2) On the WKD integration side, it's going to stip down the key anyways, but that's only a single integration
3) If we ever wanted to do extra parsing of the key info, we have that ability with the entire key

At the end of the day we run into similar problems that Facebook/Keybase/GitHub tend to fall in to that we have a choice to make.

The one these companies make is that the user is required to push updates to their keys to them. 
Systems like [mtls](https://github.com/drGrove/mtls) actually reach out keyservers to get this information. But that does leave those systems susceptible to GPG poisoning.